antool: package antool

ldorau opened the pull request:
Ref: #274

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/pmemfile/315)
<!-- Reviewable:end -->

marcinslusarz commented on pmemfile.spec:
```diff
@@ -12,6 +12,7 @@ BuildRequires:	cmake
 BuildRequires:	pkgconfig
 BuildRequires:	libpmemobj-devel
 BuildRequires:	libsyscall_intercept-devel
+BuildRequires:  python3-devel
```
s/  / /

ldorau replied:
done

marcinslusarz commented on src/tools/antool/antool:
```diff
@@ -33,8 +33,8 @@
 
-from listsyscalls import *
-from syscalltable import *
+from antoollib.listsyscalls import *
```
IMHO libantool would look better, but I don't have a strong opinion here.

ldorau replied:
In this place maybe yes, but what about the path, what is better:
```/usr/lib/python3.5/site-packages/antoollib/```
or
```/usr/lib/python3.5/site-packages/libantool/``` ?
Moreover ```antoollib``` is not a library, but helper modules for antool, so ```libantool``` could be misleading.

ldorau replied:
... so I am not sure that 'libantool' is really better

marcinslusarz replied:
Ok, I looked how other Python applications are packaged and it seems they don't install their files to /usr/lib/python*/site-packages/. What they do is put all python files in /usr/share/$appname/ and install small shell script in /usr/bin that calls python script in /usr/share/$appname/.

krzycz commented:
Reviewed 11 of 25 files at r1, 18 of 18 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/315)*
<!-- Sent from Reviewable.io -->

GBuella commented on CMakeLists.txt:
```diff
@@ -450,10 +450,23 @@ if(PKG_CONFIG_FOUND)
+if(PKG_CONFIG_FOUND)
+	pkg_check_modules(PYTHON3 QUIET python3)
+else()
+	find_package(PYTHON3 QUIET)
```
https://stackoverflow.com/questions/29245558/how-to-link-with-python3-libs-with-cmake
cmake comes with a module for findin python, so I guess these five lines could be replace by something like:
`find_package(PythonLibs 3 REQUIRED)`

Or this one to just find the interpreter:
https://cmake.org/cmake/help/v3.0/module/FindPythonInterp.html

ldorau replied:
So let's try ```find_package(PythonInterp 3 QUIET REQUIRED)``` ...

ldorau replied:
@GBuella please resolve

ldorau commented:
Review status: 4 of 11 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/tools/antool/antool.py, line 36 at r1](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsnnA0cCibXCl6ijAiK-r1-36:-KswUQmR5oIdRB8m3SPl:b-896fix) ([raw file](https://github.com/pmem/pmemfile/blob/aef5f86344c88987c910954a50f530123156c1f3/src/tools/antool/antool.py#L36)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Ok, I looked how other Python applications are packaged and it seems they don't install their files to /usr/lib/python*/site-packages/. What they do is put all python files in /usr/share/$appname/ and install small shell script in /usr/bin that calls python script in /usr/share/$appname/.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/315)*
<!-- Sent from Reviewable.io -->

ldorau commented:
Review status: 4 of 11 files reviewed at latest revision, 2 unresolved discussions.

---

*[pmemfile.spec, line 15 at r1](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsnnA0cCibXCl6ijAiH-r1-15:-KswVMaxwuFr5vYe2wov:bv3ux1l) ([raw file](https://github.com/pmem/pmemfile/blob/aef5f86344c88987c910954a50f530123156c1f3/pmemfile.spec#L15)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

done
</blockquote></details>

@marcinslusarz please resolve

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/315)*
<!-- Sent from Reviewable.io -->

GBuella commented:
Reviewed 4 of 18 files at r2, 1 of 1 files at r3, 7 of 25 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[CMakeLists.txt, line 461 at r2](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsrApH8ugFJ6M6aTz40-r2-461:-KswqoLNTrDTEgwtWUng:b-d6wobw) ([raw file](https://github.com/pmem/pmemfile/blob/616e5a3c1fcc8cd9218bcdc5d1446ef93f418f82/CMakeLists.txt#L461)):*
> if(NOT PYTHONINTERP_FOUND)

Well, since find_package is told it is REQUIRED, there should be no need to check it afterward -- if I remember correctly, find_package will abort with a fatal error if it doesn't find it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/315)*
<!-- Sent from Reviewable.io -->

ldorau commented:
Review status: 11 of 12 files reviewed at latest revision, 1 unresolved discussion.

---

*[CMakeLists.txt, line 461 at r2](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsrApH8ugFJ6M6aTz40-r2-461:-KswrvEsLiEKHXvx6v01:b-o419a3) ([raw file](https://github.com/pmem/pmemfile/blob/616e5a3c1fcc8cd9218bcdc5d1446ef93f418f82/CMakeLists.txt#L461)):*
<details><summary><i>Previously, GBuella (Gabor Buella) wrote…</i></summary><blockquote>

> if(NOT PYTHONINTERP_FOUND)

Well, since find_package is told it is REQUIRED, there should be no need to check it afterward -- if I remember correctly, find_package will abort with a fatal error if it doesn't find it.
</blockquote></details>

Right, fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/315)*
<!-- Sent from Reviewable.io -->

GBuella commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/315#-:-KswsgET9UoAeSUMA12T:bnfp4nl)*
<!-- Sent from Reviewable.io -->

marcinslusarz commented:
Reviewed 1 of 18 files at r2, 1 of 1 files at r3, 5 of 25 files at r4.
Review status: all files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

*[CMakeLists.txt, line 458 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyUU3yEEU5LBQya7Hk:-KsyUU3zg5KEE8xr1DJ7:b-6qae3m) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/CMakeLists.txt#L458)):*
> ```CMake
> endif()
> 
> find_package(PythonInterp 3 QUIET REQUIRED)
> ```

only when ANTOOL_TESTS != SKIP?

---

*[pmemfile.spec, line 18 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyAfCSdiCzKFPS7Id7:-KsyAfCSdiCzKFPS7Id8:b-yufqnt) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/pmemfile.spec#L18)):*
> ```Ruby
> BuildRequires:	libsyscall_intercept-devel
> BuildRequires:	python3
> BuildRequires:	python3-devel
> ```

Do you still need python3-devel? It's not required by Debian packages and you didn't add it to Ubuntu Dockerfiles.

---

*[pmemfile.spec, line 72 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyU3y4snV2-KFWVeAr:-KsyU3y5ERvtvYnDY94p:bmk2c1t) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/pmemfile.spec#L72)):*
> ```Ruby
> Summary: Tools for pmemfile
> Group: System Environment/Libraries
> %description -n pmemfile-tools
> ```

Requires: python3?

---

*[debian/pmemfile-tools.install, line 5 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyTaHLPSzDam90ANtA:-KsyTaHLPSzDam90ANtB:b-g2nkmf) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/debian/pmemfile-tools.install#L5)):*
> ```
> usr/share/man/man1/mkfs.pmemfile.1
> usr/bin/antool
> usr/share/antool/*
> ```

Pmemfile-tools should depend on python3. I just built this package on Debian and it was not automatically added.

---

*[utils/docker/images/Dockerfile.fedora-23, line 60 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyUitNCDg6ZmQWb3Hg:-KsyUitNCDg6ZmQWb3Hh:b3kqx) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/utils/docker/images/Dockerfile.fedora-23#L60)):*
> ```
> 	python3 \
> 	python3-coverage \
> 	python3-devel \
> ```

.

---

*[utils/docker/images/Dockerfile.fedora-24, line 60 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyUnMsfmC_X5BcmWCE:-KsyUnMsfmC_X5BcmWCF:b3kqx) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/utils/docker/images/Dockerfile.fedora-24#L60)):*
> ```
> 	python3 \
> 	python3-coverage \
> 	python3-devel \
> ```

.

---

*[utils/docker/images/Dockerfile.fedora-25, line 60 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyUoROCF4wkRQPkn0q:-KsyUoROCF4wkRQPkn0r:b3kqx) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/utils/docker/images/Dockerfile.fedora-25#L60)):*
> ```
> 	python3 \
> 	python3-coverage \
> 	python3-devel \
> ```

.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/315)*
<!-- Sent from Reviewable.io -->

ldorau commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[CMakeLists.txt, line 458 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyUU3yEEU5LBQya7Hk:-KtA_1cGLdHBc7ZZzIVq:b-7ozv9t) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/CMakeLists.txt#L458)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

only when ANTOOL_TESTS != SKIP?
</blockquote></details>

Moved to tests/antool/CMakeLists.txt

---

*[pmemfile.spec, line 18 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyAfCSdiCzKFPS7Id7:-KtA_iybHwkZB7nZTRQy:ba3dfob) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/pmemfile.spec#L18)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Do you still need python3-devel? It's not required by Debian packages and you didn't add it to Ubuntu Dockerfiles.
</blockquote></details>

Removed

---

*[pmemfile.spec, line 72 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyU3y4snV2-KFWVeAr:-KtA_lDHBuO2lrNjIzCt:b-8b11j9) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/pmemfile.spec#L72)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Requires: python3?
</blockquote></details>

Added

---

*[debian/pmemfile-tools.install, line 5 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyTaHLPSzDam90ANtA:-KtAdMHMfpOTKBpxrmxV:b-896fix) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/debian/pmemfile-tools.install#L5)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Pmemfile-tools should depend on python3. I just built this package on Debian and it was not automatically added.
</blockquote></details>

Done.

---

*[utils/docker/images/Dockerfile.fedora-23, line 60 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyUitNCDg6ZmQWb3Hg:-KtAcn8b32BJS2Ry4HcM:ba3dfob) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/utils/docker/images/Dockerfile.fedora-23#L60)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

.
</blockquote></details>

Removed

---

*[utils/docker/images/Dockerfile.fedora-24, line 60 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyUnMsfmC_X5BcmWCE:-KtAcoupCY26Y0BgDpdp:ba3dfob) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/utils/docker/images/Dockerfile.fedora-24#L60)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

.
</blockquote></details>

Removed

---

*[utils/docker/images/Dockerfile.fedora-25, line 60 at r5](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KsyUoROCF4wkRQPkn0q:-KtAcqDAZ4d10qZhekvu:ba3dfob) ([raw file](https://github.com/pmem/pmemfile/blob/ca79c1942dcaaf2b0bbde9c1f01679756621de4c/utils/docker/images/Dockerfile.fedora-25#L60)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

.
</blockquote></details>

Removed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/315)*
<!-- Sent from Reviewable.io -->

krzycz commented:
Reviewed 4 of 25 files at r4, 7 of 7 files at r6.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[src/tools/antool/antool.py, line 3 at r4](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KtBHDfQCsR3Ahs0-y2f:-KtBHDfQCsR3Ahs0-y2g:b3kre) ([raw file](https://github.com/pmem/pmemfile/blob/0654c37fde0db69d29b78440c99756365233e623/src/tools/antool/antool.py#L3)):*
> ```Python
> #!/usr/bin/python3
> #
> # Copyright (c) 2017, Intel Corporation
> ```

?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/315)*
<!-- Sent from Reviewable.io -->

marcinslusarz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 18 files at r2, 7 of 7 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/315#-:-KtBPcsSck-RpOfqOEnh:bnfp4nl)*
<!-- Sent from Reviewable.io -->

ldorau commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/tools/antool/antool.py, line 3 at r4](https://reviewable.io:443/reviews/pmem/pmemfile/315#-KtBHDfQCsR3Ahs0-y2f:-KtC4_1RbEcN6GU-b1XC:b-rcq008) ([raw file](https://github.com/pmem/pmemfile/blob/0654c37fde0db69d29b78440c99756365233e623/src/tools/antool/antool.py#L3)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

?
</blockquote></details>

It is already fixed on the master

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/315)*
<!-- Sent from Reviewable.io -->
